great_expectations/execution_engine/execution_engine.py

```diff
@@ -189,6 +200,27 @@ class SplitDomainKwargs:
 
 
 class ExecutionEngine(ABC):
+    """ExecutionEngine defines interfaces and provides common methods for loading Batch of data and compute metrics.
+
+    ExecutionEngine is the parent class of every backend-specific computational class, tasked with loadsing Batch of
+    data and computing metrics.  Each subclass (corresponding to Pandas, Spark, SQLAlchemy, and any other computational
+    components) utilizes generally-applicable methods defined herein, while implementing required backend-specific
+    interfaces.  ExecutionEngine base class also performs the important task of translating cloud storage resource URLs
+    to format and protocol compatible with given computational mechanism (e.g, Pandas, Spark).  Then ExecutionEngine
+    subclasses access the referenced data according to their paritcular compatible protocol and return Batch of data.
+
+    In order to obtain Batch of data, ExecutionEngine (through implementation of key interface methods by subclasses)
+    gets data records and also provides access references so that different aspects of data can be loaded at once.
+    ExecutionEngine uses BatchManager for Batch caching in order to reduce load on computational backends.
+
+    Crucially, ExecutionEngine serves as focal point for resolving (i.e., computing) metrics.  Wherever opportunities
+    arize to bundle multiple metric computations (e.g., SQLAlchemy, Spark), ExecutionEngine utilizes subclasses in order
+    to provide specific functionality (bundling of computation is available only for "deferred execution" computational
+    systems, such as SQLAlchemy and Spark; it is not available for Pandas, because Pandas computations are immediate).
+
+    Finally, ExecutionEngine defines interfaces for Batch data sampling and splitting Batch of data along defined axes.
```

Comment:
Can you add an `Args:` section here with the arguments that can be passed into the constructor?